LBA v2.22 Light Deployable.txt

```diff
@@ -1,339 +1,346 @@
+                if(target==myKey||target==myOwner)//First things first, am I the target?
+                {
+                    key owner=llList2Key(ownerinfo,0);//Gets the owner key from Ownerinfo
+                    key creator=llList2Key(ownerinfo,1);//Gets the creator key from Ownerinfo
+                    integer att=llList2Integer(ownerinfo,2);
+                    string n2=llKey2Name(owner);//Gets the owner name
+                    if(n2=="")n2=owner;//If you can't get the owner name, return the key instead
+                    integer dmg=(integer)llGetSubString(m,37,-1);//This is the damage, fuck you.
+                    if(att)
+                    {
+                        list tdl=llGetObjectDetails(owner,[OBJECT_POS,OBJECT_ROT]);//target data list
+                        vector csize=llGetAgentSize(owner);//Gets hitbox size for camera position adjustment
+                        vector tpos=llList2Vector(tdl,0)+<0,0,csize.z/2>;//Their pos
+                        rotation trot=llList2Rot(tdl,1);//Their rot
+                        vector camvec=tpos+<1,0,0>*trot*llVecDist(tpos,llGetPos());
+                        camvec=(camvec-llGetPos())/llGetRot();
+                        integer bc=0;
+                        if (camvec.x>min.x&&camvec.y>min.y&&camvec.z>min.z&&
+                        camvec.x<max.x&&camvec.y<max.y&&camvec.z<max.z)bc=1;
+                        if(!bc)
+                        {
+                            return;
+                        }
+                    }
+                    if(!att)
+                    {
+                        if(dmg>1)
+                        {
+                            if(llListFindList(recent,[id])>-1)return;//Check if the object has already damaged you
+                            else recent+=id;
+                        }
+                    }
+                    if(dmg>5)llRegionSay(reghex*-3,"log:"+(string)owner+","+(string)creator+","+(string)n+","+(string)dmg);//For logging hits within the region for later evaluation. If you don't have this active, blacklisting won't even work, man.
+                    proc+=[owner,n2,creator,n,dmg];//Adds to the processing event
+                    ++events;//Adds to events
+                    if(events==1)llSetTimerEvent(1*llGetRegionTimeDilation());//On the first event, the processing countdown/timer gets started.
+                }
+            }
+        }
+    }
+    timer()
+    {
+        integer i=0;
+        while(i<events)
+        {
+            key owner=llList2Key(proc,0);
+            string ownern=llList2String(proc,1);
+            key creator=llList2Key(proc,2);
+            string name=llList2Key(proc,3);
+            integer dmg=llList2Integer(proc,4);
+            integer full=dmg;
+            if(atcap>0)
+            {
+                if (atcap) // use atcap to limit repairs
```

Comment:
Lines 229 to 236: This is not the intended implementation of the fix. These lines should be moved up and inserted after line 183.